### PR TITLE
schema -> typeDefs

### PR DESCRIPTION
**Pull request by ecwyne**
Currently this warns but works for data sources providing `schema` instead of `typeDefs`

I'll encourage the use of "Type Definitions" and "where GraphQL types are defined" in the docs as opposed to "Schema" to help avoid confusion

### CODE REVIEW

File: src/gramps.js [hunk 1]
```diff
@@ -33,23 +33,31 @@ const getDefaultApolloOptions = options => ({
...
       if (mock) {
         addMockFunctionsToSchema({
-          schema,
+          schema: executableSchema,
           mocks,
           ...options.addMockFunctionsToSchema,
         });
       }
-      return schema;
+      return executableSchema;
```
Comment (jlengstorf) [inline src/gramps.js]: Code Climate is complaining about complexity. I think we can probably extract some of this (for example, the typeDefs check) into a helper function.

File: src/rootSource.js [hunk 2]
```diff
@@ -1,6 +1,21 @@
...
+
+type Mutation {
+  # Returns a charming message from GrAMPS.
+  grampsPing: String!
+}
+
+schema {
+  query: Query
+  mutation: Mutation
+}
```
Comment (jlengstorf) [inline src/rootSource.js]: This is _super_ nitpicky, but can we indent this?

**Comment by ecwyne** [comment]
Is there any way to test codeclimate locally? I'd love to find out when it complains about complexity before I push.

**Comment by jlengstorf** [comment]
@ecwyne This should work: https://github.com/codeclimate/codeclimate

I believe the ESLint/Prettier rules _should_ catch this as well. Those ought to be running pre-push, assuming `husky` installed properly on your machine. If the complexity checks _aren't_ working, we should open an issue so I can figure out how to get it working. There are definitely ESLint rules to check it (cyclomatic complexity et al.).

File: src/gramps.js [hunk 3]
```diff
@@ -24,6 +24,19 @@ const getDefaultApolloOptions = options => ({
...
+    console.warn(
+      namespace,
+      'Type definitions must be exported as "typeDefs".',
+      'Use of "schema" has been deprecated and will be removed in a future release',
+    );
+    return typeDefs || schema;
+  } else {
+    return typeDefs;
+  }
+};
```
Comment (jlengstorf) [inline src/gramps.js]: Would it work to do this:
```js
const getTypeDefs = ({ schema, typeDefs, namespace }) => {
  if (typeof schema === 'string') {
    console.warn(
      namespace,
      'Type definitions must be exported as "typeDefs".',
      'Use of "schema" has been deprecated and will be removed in a future release',
    );
  }

  return typeDefs || schema;
};
```

This gives us the same result, I believe, and ensures we get a return.

**Comment by ecwyne** [comment]
@jlengstorf this PR should be good to go.

I do have husky/eslint set up properly on my machine. I'm getting linting in my editor (sublime text) and everything and husky is running pre-push. 

I don't see anything about cyclomatic complexity in the eslint rules though, unless it's a part of airbnb-base or prettier?

**Comment by jlengstorf** [comment]
Hmm... I'll check out ESLint. I could have sworn I pulled those rules down, but it must have been in another project.

**Pull request merged**